dataprofiler/data_readers/csv_data.py

```diff
@@ -148,26 +149,28 @@ def _guess_header_row(data_as_str, suggested_delimiter=None,
                     cell_type = 'none'
                 else:
             
-                    if cell.isdigit():
-                        cell_type = 'int'
-                        
-                    try:
-                        float(cell)
-                        cell_type = 'float'
-                    except ValueError:
-                        pass
-
                     try: 
-                        if dateutil.parser.parse(cell):
+                        if dateutil.parser.parse(cell, fuzzy=False):
                             cell_type = 'date'
                     except ValueError:
                         pass
                     except OverflowError:
                         pass
+
+                    try:
+                        float(cell)
+                        cell_type = 'float'
+                    except ValueError:
+                        pass
+
+                    if cell.isdigit():
+                        cell_type = 'int'
```

Comment:
integers have to come last as integers can trigger prior try/except statements.

Reply:
I'd suggest using if cell_type=='float', then check if cell == int(float(cell)), cause otherwise, this doesn't work `'-123.isdigit()'`